tests/integration/test_non_user_endpoints.py

```diff
@@ -240,6 +241,13 @@ def test_search_timeout(self):
         self.assertRaises((requests.exceptions.Timeout, requests.exceptions.ConnectionError),
                           lambda: sp.search(q='my*', type='track'))
 
+    def test_max_retries_reached(self):
+        client_credentials_manager = SpotifyClientCredentials()
+        sp = spotipy.Spotify(client_credentials_manager=client_credentials_manager)
+        self.assertRaises((spotipy.exceptions.SpotifyException,),
+                          lambda: sp.playlist_is_following(playlist_id=AuthTestSpotipy.playlist_id,
+                                                           user_ids=[AuthTestSpotipy.bad_id]))
```

Comment:
Your test doesn't seem to test the retry code. I had a look and came up with the following:


```suggestion
        spotify_no_retry = Spotify(
            client_credentials_manager=SpotifyClientCredentials(),
            retries=0)
        i = 0
        while i < 100:
            try:
                spotify_no_retry.search(q='foo')
            except spotipy.exceptions.SpotifyException as e:
                self.assertIsInstance(e, spotipy.exceptions.SpotifyException)
                return
            i += 1
        self.fail()
```

it's not ideal as it's doing many calls without mocking, but it should do for now

Reply:
Fair point, I was going for the 'many-call' approach but in retrospect I agree a mock-based approach is better.